src/main/java/org/threeten/extra/chrono/InternationalFixedDate.java

```diff
@@ -0,0 +1,965 @@
+
+    long getProlepticWeek() {
+        return ((long) prolepticYear) * InternationalFixedChronology.WEEKS_IN_YEAR +
+                getCalculatedMonth() * InternationalFixedChronology.WEEKS_IN_MONTH +
+                ((getCalculatedDayOfMonth() - 1) / InternationalFixedChronology.DAYS_IN_WEEK) - 1;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public InternationalFixedDate minus(final TemporalAmount amount) {
+        return (InternationalFixedDate) amount.subtractFrom(this);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public InternationalFixedDate minus(final long amountToSubtract, final TemporalUnit unit) {
+        return (amountToSubtract == Long.MIN_VALUE ? plus(Long.MAX_VALUE, unit).plus(1, unit) : plus(-amountToSubtract, unit));
+    }
+
+    //-------------------------------------------------------------------------
+    /**
+     * {@inheritDoc}
+     */
+    @Override  // for covariant return type
+    @SuppressWarnings ("unchecked")
+    public ChronoLocalDateTime<InternationalFixedDate> atTime(final LocalTime localTime) {
+        return (ChronoLocalDateTime<InternationalFixedDate>) ChronoLocalDate.super.atTime(localTime);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public long until(final Temporal endExclusive, final TemporalUnit unit) {
+        return until(InternationalFixedDate.from(endExclusive), unit);
+    }
+
+    long until(final InternationalFixedDate end, final TemporalUnit unit) {
+        if (unit instanceof ChronoUnit) {
+            switch ((ChronoUnit) unit) {
+                case WEEKS:
+                    return weeksUntil(end);
+                case MONTHS:
+                    return monthsUntil(end);
+                default:
+                    break;
+            }
+        }
+
+        return super.until(end, unit);
+    }
+
+    /**
+     * Get the number of years from this date to the given day.
+     *
+     * @param end The end date.
+     * @return The number of years from this date to the given day.
+     */
+    private long yearsUntil(final InternationalFixedDate end) {
+        long startYear = getProlepticYear() * 512L + getDayOfYear();
+        long endYear = end.getProlepticYear() * 512L + end.getDayOfYear();
+
+        return (endYear - startYear) / 512L;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public ChronoPeriod until(final ChronoLocalDate endDateExclusive) {
+        InternationalFixedDate end = InternationalFixedDate.from(endDateExclusive);
+        int years = Math.toIntExact(yearsUntil(end));
+        // Get to the same "whole" year.
+        InternationalFixedDate sameYearEnd = (InternationalFixedDate) end.plusYears(years);
+        int months = (int) monthsUntil(sameYearEnd);
+        int days = (int) daysUntil(sameYearEnd.plusMonths(months));
+
+        return getChronology().period(years, months, days);
+    }
+
+    private long weeksUntil(final InternationalFixedDate fixed) {
```

Comment:
`weeksUntil(...)` - parameter name should be `end`, `fixed` is confusing.  The calculation sometimes returns incorrect results: 2012 Year Day weeks until 2016 Year Day yields 52 * 4 - 1, not the expected 4 complete years.  Possibly year days in similar situations (haven't run those numbers).  I _think_ you have to offset whenever either the start or end is a different flag value than the other, or one is a flag value and the other isn't, but I don't have relevant test cases at the moment.  At minimum, you shouldn't be offsetting if both dates are the same leap-day/year-day.  Add the missing test data for this, and correct the implementation.